### ISSUE TITLE
Netedit: Can't use select tool when decals are loaded

### ISSUE THREAD

**Issue by jmlouw** [eclipse/sumo#13033]
After I've loaded a background image I can't use the select tool to select elements of the network. I've tried `.png` and `.jpg` images for the background. Only after removing the `decal` I can select elements again. I only recently started to experience this problem. I've tried multiple versions of SUMO and reinstalling dependencies. 

![image](https://user-images.githubusercontent.com/42248038/231982011-0fe2a96c-f7b7-4b58-a1d0-b4b7e6a464e0.png)
In the above screenshot I'm trying to select the edge in `inspect mode` by clicking on the edge, but it does not get selected.

My decal set up:
![image](https://user-images.githubusercontent.com/42248038/231982397-b404f381-943a-4942-90d0-69ecf0660eba.png)

Selecting the edge after background is removed:
![image](https://user-images.githubusercontent.com/42248038/231982578-f43ee53e-d2d1-4d36-b7e4-15f3bb9b34d2.png)

As you can see I have no trouble selecting the edge after the background is removed. 

**SUMO-version:**
```
Eclipse SUMO sumo Version 1.16.0
 Build features: Linux-5.4.0-137-generic x86_64 GNU 11.3.0 Release FMI Proj GUI Intl SWIG GDAL GL2PS Eigen
 Copyright (C) 2001-2023 German Aerospace Center (DLR) and others; https://sumo.dlr.de
```

**operating system:**
```
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.2 LTS
Release:	22.04
Codename:	jammy
```

**Comment by jmlouw** [comment]
In general I've experience problems selecting nodes and different elements in Netedit. For example selecting lanes when setting up connection in a junction is very buggy and elements are only selected when clicking on certain parts
![Screencast from 14-04-2023 10_18_43](https://user-images.githubusercontent.com/42248038/231987231-31026023-fde3-40dd-a7a7-2be02c108200.gif)

**Comment by palvarezlopez** [comment]
@jmlouw Thanks!

**Comment by namdre** [comment]
I cannot reproduce it. Even if the background image has a higher layer than the network elements (thus hiding them), selection mode still picks up the network elements.

**Comment by namdre** [comment]
> In general I've experience problems selecting nodes and different elements in Netedit. In the example below I'm trying to select nodes to join as a junction, but the top one keeps selecting the edge instead of the node.

whenever you click on overlapping objects, you should be able to cycle through the list by repeatedly clicking on the same spot.

**Comment by jmlouw** [comment]
After playing around a bit more. I'm experiencing the problem both with and without the background image. I've tried a fresh Ubuntu installation as well as a fresh SUMO installation. I'm only experiencing the problem on Ubuntu and not on my Mac. In the GIF I've provided I constantly click on the elements, but they only get selected with certain clicks. So definitely feels like a bug - I can cycle through overlapping objects on my Mac. 

I've tried building from source (multiple versions) as well as using the package manager i.e. `sudo apt-get install sumo sumo-tools sumo-doc`. It feels that it may be an underlying dependency that is causing the problem as I've only started to experience it recently (Perhaps after `sudo apt-get update and sudo apt-get upgrade`, but this is just me guessing)

I'm not sure what next steps are?
- Is there anything else I can provide?
- What is the recommended method to install?

**Comment by behrisch** [comment]
Can you try the [pip installed sumo](https://sumo.dlr.de/docs/Downloads.php#python_packages_virtual_environments)? It brings some libraries with it so we can at least rule out some dependencies.

**Comment by jmlouw** [comment]
I tried `pip install sumo` and ` pip install eclipse-sumo`, but still experiencing the problem. Definitely more severe when a background image is loaded, since I then can't select anything. 

Uploaded this example to Youtube to illustrate the problem. I've included `screemkey` to show my left mouse clicks. https://www.youtube.com/watch?v=Ci37jrkZM5E&ab_channel=Quebitransport

**Comment by behrisch** [comment]
did you purge all dependencies before doing `pip install eclipse-sumo`?

**Comment by jmlouw** [comment]
I'm not sure what the best way is to purge all dependencies? I used `pip-autoremove` to uninstall `eclipse-sumo`. I've created a fresh virtual environment and reinstall `eclipse-sumo`. The problem persists no matter the version of sumo I install. I've tried `1.16.0`, `1.14.1` and `1.8.0`.

**Comment by behrisch** [comment]
No, I meant uninstalling things like libproj, libFox and libxerces before doing the pip install. Or just trying a clean ubuntu. Do you have an unusual graphics card or is your ubuntu running wayland?

**Comment by jmlouw** [comment]
I can definitely try a clean Ubuntu. I have a Dell XPS 13 9310 and I'm running Wayland, but tried switching to xorg.

**Comment by jmlouw** [comment]
Fresh install of Ubuntu on a new PC, still the same issue - can't select any of the elements after importing the image.

```
Distributor ID:    Ubuntu
Description:    Ubuntu 22.04.2 LTS
Release:    22.04
Codename:    jammy
Python 3.8.16
```

![sample](https://user-images.githubusercontent.com/42248038/233550824-537b72f4-a3cf-45e3-94e0-3f5c7d9fa4f3.gif)

**Comment by behrisch** [comment]
again with wayland and xorg?

**Comment by jmlouw** [comment]
Switched to xorg by logging out and selecting `Ubuntu on Xorg`, so the problem is both on xorg and wayland.

**Comment by jmlouw** [comment]
This time I created a new virtual env and installed by using: `pip install eclipse-sumo`. This is the only command I ran.

**Comment by behrisch** [comment]
I am clueless. We are testing regularly on ubuntu 22.04.2, I have two machines running this OS and never encountered this. My guess is that it is something about the graphics chip (my test machines have nvidia) but that is really just a wild guess. Can you try to disable any optional features like 3D acceleration?

**Comment by jmlouw** [comment]
Okay so I got it fixed on one of my computers. I updated my Intel graphics drivers using this [guide](https://dgpu-docs.intel.com/installation-guides/ubuntu/ubuntu-jammy-arc.html). So it's definitely related to the graphics drivers. 

I don't know if there is a better way to upgrade the drivers, because I'm facing other problems like not being able to choose Wayland or Xorg when logging in and I can't change the resolution of my display.

**Comment by behrisch** [comment]
Good to hear that and happy SUMOing :-)

**Comment by jmlouw** [comment]
So I made a mistake and installed accidentally installed Arc drivers on my machine (I don't have an Arc GPU). My computer was sort of bricked... 

So then decided to rather do a fresh install of Ubuntu 20.04. I have **no problem** with Netedit in that version of Ubuntu at all. So then I thought to try to upgrade to Ubuntu 22,04.2 LTS. I chose the option to remove unused dependencies. 

I now again have the problem in 22.04.2 LTS. I also recall that the first time I experienced the problem was after running `sudo apt-get autoremove`.

So is someone perhaps willing to run `sudo apt-get autoremove` in Ubuntu 22.04.2 and see if that recreates the problem? I'm pretty sure that an important dependency gets removed and causes the problem. Otherwise the problem is specific to my hardware.

**Comment by jmlouw** [comment]
Just a last update for anyone that is also facing this issue. The problem definitely feels graphics driver related and seems that it is related to Intel integrated graphics (In my case Xe Graphics). So here is a list of things I tried:
✅ - no problem, Netedit GUI is working as it should
❌ - cannot interact with elements on GUI

✅ `Ubuntu 20.04.6 LTS (Focal Fossa)` 
❌ `Ubuntu 22.04.2 LTS (Jammy Jellyfish)` 
❌ `Ubuntu 23.04 (Lunar Lobster)`
✅ `Ubuntu 22.04.2 LTS (Jammy Jellyfish)`  with Intel Arc drivers (fixed the SUMO issue, but caused a lot of other problems)

I also tried launching it with i3, Xorg, and Wayland but did not make a difference. I also tried upgrading for `20.04.6` to `22.04.2` and keep dependencies of  `20.04.6` , but made no difference. I also tried ticking the install third drivers with installation, but no difference.

So my final thoughts. It's an Intel Xe driver issue from Ubuntu 22.02.2 onwards. I don't know how to change or upgrade the drivers. So for now I'm downgrading to `Ubuntu 20.04.6 LTS (Focal Fossa)`  to fix the issue.
